File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file294.f90
subroutine init_array(n, a, b)
implicit none

DATA_TYPE, dimension(n) :: a
DATA_TYPE, dimension(n) :: b
integer :: n
integer :: i

do i = 1, n
a(i) = (DBLE(i-1) + 2.0D0) / n
b(i) = (DBLE(i-1) + 3.0D0) / n
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file294.cpp
void init_array(int n, double* a, double* b){
 for (int i=0; i<n; i++){
 a[i] = (double(i-1) + 2.0) / n;
 b[i] = (double(i-1) + 3.0) / n;
 }
}
